internal/pkg/cli/dialog/use_template.go

```diff
@@ -5,120 +5,185 @@ import (
 	"fmt"
 	"strconv"
 
+	"github.com/keboola/keboola-as-code/internal/pkg/cli/options"
 	"github.com/keboola/keboola-as-code/internal/pkg/cli/prompt"
+	"github.com/keboola/keboola-as-code/internal/pkg/project"
 	"github.com/keboola/keboola-as-code/internal/pkg/template/input"
-	"github.com/keboola/keboola-as-code/internal/pkg/utils"
+	useTemplate "github.com/keboola/keboola-as-code/pkg/lib/operation/project/local/template/use"
+	loadState "github.com/keboola/keboola-as-code/pkg/lib/operation/state/load"
 )
 
 type contextKey string
 
-func (p *Dialogs) AskUseTemplateOptions(inputs *input.Inputs) (results map[string]interface{}, err error) {
-	results = make(map[string]interface{})
-	ctx := context.Background()
-	errors := utils.NewMultiError()
-	for _, i := range inputs.All() {
-		if !i.Available(results) {
+type useTmplDialogDeps interface {
+	Options() *options.Options
+	ProjectState(loadOptions loadState.Options) (*project.State, error)
+}
+
+type useTmplDialog struct {
+	*Dialogs
+	deps             useTmplDialogDeps
+	loadStateOptions loadState.Options
+	out              useTemplate.Options
+	context          context.Context        // for input.ValidateUserInput
+	inputsValues     map[string]interface{} // for input.Available
+}
+
+func (p *Dialogs) AskUseTemplateOptions(inputs *input.Inputs, d useTmplDialogDeps, loadStateOptions loadState.Options) (useTemplate.Options, error) {
+	return (&useTmplDialog{
+		Dialogs:          p,
+		deps:             d,
+		loadStateOptions: loadStateOptions,
+		context:          context.Background(),
+		inputsValues:     make(map[string]interface{}),
+	}).ask(inputs)
+}
+
+func (d *useTmplDialog) ask(inputs *input.Inputs) (useTemplate.Options, error) {
+	// Load state
+	projectState, err := d.deps.ProjectState(d.loadStateOptions)
+	if err != nil {
+		return d.out, err
+	}
+
+	// Target branch
+	targetBranch, err := d.SelectBranch(d.deps.Options(), projectState.LocalObjects().Branches(), `Select the target branch`)
+	if err != nil {
+		return d.out, err
+	}
+	d.out.TargetBranch = targetBranch.BranchKey
+
+	// User inputs
+	if err := d.askInputs(inputs); err != nil {
+		return d.out, err
+	}
+
+	return d.out, nil
+}
+
+func (d *useTmplDialog) addInputValue(inputDef input.Input, value interface{}) error {
+	if err := inputDef.ValidateUserInput(value, d.context); err != nil {
+		return err
+	}
```

Comment:
`addInputValue` obsahuje validaciu vstupu, ... bolo to povodne duplicitne pre kazdy input type.
V dalsom PR pridam moznost zadat uzivatelske vstupy ako `inputs.json`, a tam to tiez vyuzijem.